docs/Modules/common-modules/event-bus.md

```diff
@@ -9,4 +9,174 @@ parent: "Common Modules"
 grand_parent: "Modules"
 ---
 
-# eventBus
\ No newline at end of file
+# eventBus
+
+A general purpose event bus.
+
+It is used to communicate across a diagram instance.
+Several components of a diagram can use it to listen to and to broadcast events.
+
+It provides the following methods:
+
+* [on](#on)
+* [once](#once)
+* [off](#off)
+* [createEvent](#createEvent)
+* [fire](#fire)
+* [handleError](#handleError)
+
+## on
+
+### Usage
+
+on(events: _String_ or _Array< string >_, [priority=1000]: _number_, callback: _Function_, [that]:_Object_): _undefined_
+
+### Parameters
+
+* `events`: name of the event or _Array_ of event names.
+* `priority`: optional parameter specifying the priority in which this listener is called, where a higher number represents a higher priority and therefore leads to earlier being called. If not provided, default of `1000` will be used.
+* `callback`: function which will be invoked with `event, ...additionalArguments`. Returning anything but `undefined` from this will stop the listener propagation. Returning `false` will prevent the execution of the default action for that event.
+*  `that`: optional parameter which can be used to pass context (`this`) to the callback.
+
+### Return Value
+
+_undefined_
+
+### Code Example
+
+```javascript
+eventBus.on('saveSVG.start', 1500, function(event) {
+
+  // access event type
+  event.type; // 'saveSVG.start'
+
+  // stop propagation to other listeners
+  event.stopPropagation();
+
+  // prevent event default
+  event.preventDefault();
+
+  // Some custom logic
+  // [...]
+});
+```
+
+
+## once
+
+### Usage
+
+once(events: _String_ or _Array< string >_, [priority=1000]: _number_, callback: _Function_, [that]:_Object_): _undefined_
+
+### Parameters
+
+* `events`: name of the event or _Array_ of event names.
+* `priority`: optional parameter specifying the priority in which this listener is called, where a higher number represents a higher priority and therefore leads to earlier being called. If not provided, default of `1000` will be used.
+* `callback`: function which will be invoked with `event, ...additionalArguments`. Returning anything but `undefined` from this will stop the listener propagation. Returning `false` will prevent the execution of the default action for that event.
+*  `that`: optional parameter which can be used to pass context (`this`) to the callback.
+
+### Return Value
+
+_undefined_
+
+### Code Example
+
+```javascript
+let count = 0;
+
+function countListener() {
```

Comment:
This example is quite hard to understand and I'm not sure how it's supposed to work.